### PR TITLE
Guard Py_mod_gil under Py_GIL_DISABLED in code example

### PR DESCRIPTION
**Pull request by lysnikolaou**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,15 +85,19 @@ Extensions that use single-phase initialization need to call
...
+PyInit__module(void)
+{
+    PyObject *mod = PyModule_Create(&module);
+    if (mod == NULL) {
+        return NULL;
+    }
 
 #ifdef Py_GIL_DISABLED
-PyModule_ExperimentalSetGIL(mod, Py_MOD_GIL_NOT_USED);
+    PyModule_ExperimentalSetGIL(mod, Py_MOD_GIL_NOT_USED);
```
Comment (rgommers) [inline README.md:98]: Looks like this should be updated to `PyUnstable_Module_SetGIL`

**Review by rgommers — COMMENTED** [review]
Thanks @lysnikolaou. The added guards look good - I spotted one other issue

File: README.md (hunk 1 above)
Comment (lysnikolaou) [reply README.md:98]: Right. Thanks @rgommers!

**Review by rgommers — APPROVED** [review]
LGTM now, in it goes. Thanks Lysandros

**Pull request merged**